hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/pipeline/ComputeStageImplBase.java

```diff
@@ -223,6 +223,7 @@
             @Nonnull BiFunctionEx<? super S, ? super T, ? extends R> mapFn
     ) {
         checkSerializable(mapFn, "mapFn");
+        pipelineImpl.attachFiles(serviceFactory.attachedFiles());
```

Comment:
we should do this another way - it's easy to miss in one place to add the files, and you don't have coverage for all the options. Maybe Pipeline.attachFiles() or stage.addFiles() ?